client-app/app/components/page-nav.js

```diff
@@ -0,0 +1,45 @@
+import Component from "@ember/component";
+import { computed, observer } from "@ember/object";
+import { later } from "@ember/runloop";
+
+export default Component.extend({
+  tagName: "div",
+  classNames: ["nav-controls"],
+  classNameBindings: ["extraClasses"],
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+    later(this, () => {
```

Comment:
Multiple notes here:
- layer without a delay is a smell to me, it reads like "couldn't make it work so I added a small delay and it works". :p wouldn't and afterRender be better?
- you should cancel any later in a willDestroyElement, otherwise this might create error when the component gets destroyed and later hasn't occured yet
- this could be inline: `later(this, () => this.navigate(this.position))`

Reply:
A big chunk of the code in this PR is about 1 year old, including this override to the `didReceiveAttrs` method. I've removed this override entirely since I've refactored this component and the override is no longer needed 👍